discoverd/deployment/deployment.go

```diff
@@ -88,6 +89,21 @@ outer:
 		if err := d.update(); err != nil {
 			return err
 		}
+		// If the metadata is nil then wait for another metadata event before re-evaluating
+		if d.meta == nil {
+			select {
+			case event, ok := <-d.events:
+				if !ok {
+					return fmt.Errorf("service stream closed unexpectedly: %s", d.stream.Err())
+				}
+				if event.Kind == discoverd.EventKindServiceMeta {
+					d.meta = event.ServiceMeta
+					continue outer
```

Comment:
The difference here is we are just making sure a service meta event has been received. `update` only ensures that the events channel has been drained. Ensuring that a service meta even has been received should ensure that decode shouldn't fail (as apart from when no deployment has started we never leave the metadata in a state that is not decodable by `decode`).